qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1517A.py

```diff
@@ -287,3 +671,151 @@ def use_high_resolution_adc(self) -> None:
+                                int] = constants.Abort.ENABLED,
+            sweep_mode: Union[constants.SweepMode,
+                                int] = constants.SweepMode.LINEAR
+
+    ) -> None:
+        """
+        Setup the staircase sweep measurement using the same set of commands
+        (in the same order) as given in the programming manual - see pages
+        3-19 and 3-20.
+
+        Args:
+            v_start: starting voltage of staircase sweep
+            v_end: ending voltage of staircase sweep
+            n_steps: number of measurement points (uniformly distributed
+                between v_start and v_end)
+            post_sweep_voltage_val: voltage to hold at end of sweep (i.e.
+                start or end val). Sweep chan will also output this voltage
+                if an abort condition is encountered during the sweep
+            av_coef: coefficient to use for av command to set ADC
+                averaging.  Negative value implies NPLC mode with absolute
+                value of av_coeff the NPLC setting to use. Positive value
+                implies auto mode and must be set to >= 4
+            enable_filter: turn SMU filter on or off
+            v_src_range: range setting to use for voltage source
+            i_comp: current compliance level
+            i_meas_range: current measurement range
+            hold_time: time (in s) to wait before starting very first
+                measurement in sweep
+            delay: time (in s) after starting to force a step output and
+                before starting a step measurement
+            step_delay: time (in s) after starting a step measurement before
+                next step in staircase. If step_delay is < measurement time,
+                B1500 waits until measurement complete and then forces the
+                next step value.
+            measure_delay: time (in s)  after receiving a start step
+                measurement trigger and before starting a step measurement
+            abort_enabled: Enbale abort
+            sweep_mode: Linear, log, linear-2-way or log-2-way
+          """
+        self.set_average_samples_for_high_speed_adc(av_coef)
+        self.connection_mode_of_smu_filter(enable_filter=enable_filter)
+        self.source_config(output_range=v_src_range,
+                           compliance=i_comp,
+                           min_compliance_range=i_meas_range)
+        self.voltage(v_start)
+
+        self.measurement_operation_mode(constants.CMM.Mode.COMPLIANCE_SIDE)
+        #only for cureent channel
+
+        self.current_measurement_range(i_meas_range)
+
+        self.iv_sweep.hold_time(hold_time)
+        self.iv_sweep.delay(delay)
+        self.iv_sweep.step_delay(step_delay)
+        self.iv_sweep.measure_delay(measure_delay)
+        self.iv_sweep.sweep_auto_abort(abort_enabled)
+        self.iv_sweep.post_sweep_voltage_condition(post_sweep_voltage_val)
+        self.iv_sweep.chan(self.channels[0])
```

Comment:
this should not be needed.... let's try to see why it was needed in the first testing?

Reply:
Not needed any more. This got fixed

Reply:
Well I was wrong. Testing again today, I reached following conclusion. 

Unfortunately  setup_staircase_sweep, self.iv_sweep.chan(self.channels[0]) without this line it is still giving me error. The reason is that I am getting response of sweep settings from from the instrument. The catch is that sweep setting can be implemented on only one channel. First SMU is initialised , the sweep setting become 'wv1,2,....' then smu 2 is initialized and hence sweep settings become 'wv2,1,...'. Now while setting  sweep_steps in smu1, smu1.iv_sweep.sweep_steps(201), group parameter will first get all the sweep settings from the instrument. It will hence get channel =2. And here is our problem.

Reply:
Reliazed that GroupParameter may not be best for this. Instead went to old way of defining get set cmd for each parameter and storing the state of all other coupled parameters in a dictionary.